blockstack_client/keys.py

```diff
@@ -516,27 +490,23 @@ def get_pubkey_addresses( pubkey ):
 
     Return (compressed address, uncompressed address)
     """
-    public_key_object = ECPublicKey(pubkey, version_byte=virtualchain.version_byte)
-    compressed_address = None
-    uncompressed_address = None
-
-    if public_key_object._type == PubkeyType.compressed:
-        compressed_address = public_key_object.address()
-        uncompressed_address = bin_hash160_to_address(
-            bin_hash160(
-                decompress(public_key_object.to_bin())
-            ),
-            version_byte=virtualchain.version_byte
-        )
-    elif public_key_object._type == PubkeyType.uncompressed:
-        compressed_address = bin_hash160_to_address(
-            bin_hash160(
-                compress(public_key_object.to_bin())
-            ),
-            version_byte=virtualchain.version_byte
-        )
-        uncompressed_address = public_key_object.address()
+    version_byte = virtualchain.version_byte
+    compressed_address, uncompressed_address = None, None
+
+    pubkey = ECPublicKey(pubkey, version_byte=version_byte)
+    pubkey_bin = pubkey.to_bin()
+
+    if pubkey._type == PubkeyType.compressed:
+        compressed_address = pubkey.address()
+        uncompressed_address = decompress(pubkey_bin)
+        hashed_address = bin_hash160(uncompressed_address)
+        uncompressed_address = bin_hash160_to_address(hashed_address, version_byte=version_byte)
+    elif pubkey._type == PubkeyType.uncompressed:
+        uncompressed_address = pubkey.address()
+        compressed_address = compress(pubkey_bin)
+        hashed_address = bin_hash160(compressed_address)
+        compressed_address = bin_hash160_to_address(hashed_address, version_byte=version_byte)
     else:
-        raise Exception("Invalid public key")
+        raise Exception('Invalid public key')
 
-    return (compressed_address, uncompressed_address)
+    return compressed_address, uncompressed_address
```

Comment:
Looks good; thank you for cleaning this up!  I will also be testing this one rigorously, since it affects how the user's wallet gets loaded and stored.